picdescbot/mastodon.py

```diff
@@ -0,0 +1,41 @@
+# coding=utf-8
+# picdescbot: a tiny twitter/tumblr bot that tweets random pictures from wikipedia and their descriptions
+# this file implements Mastodon-related functionality
+# Copyright (C) 2017 Federico Mena Quintero <federico@gnome.org>
+
+# This uses Mastodon.py: https://pypi.python.org/pypi/Mastodon.py/1.0.5
+# Documentation for that: https://mastodonpy.readthedocs.io/en/latest/
+
+from mastodon import Mastodon
+
+class Client(object):
+    name = "mastodon"
+
+    # FIXME: Note how
+    # https://mastodonpy.readthedocs.io/en/latest/#app-registration-and-user-authentication
+    # has the code bit to generate our mastodon-clientcred.txt.  We
+    # don't do this here; do it by hand!
+    #
+    # Mastodon.create_app("picdescbot",
+    #                     scopes       = ["write"],    # no read or follow
+    #                     api_base_url = "https://botsin.space",
+    #                     to_file      = "mastodon-clientcred.txt")
+    #
+
+    def __init__(self, config):
+        # FIXME: pull this out of `config`
+        self.mastodon = Mastodon(client_id = "mastodon-clientcred.txt")
+
+        # FIXME: pull these out of `config`
+        self.access_token = mastodon.log_in("username@example.com",
+                                            "supersecretpassword")
+
+    def send(self, picture):
+        filename = picture.url.split('/')[-1]
+        data = picture.download_picture()
+
+        posted_pic_ids = self.mastodon.media_post (filename)
```

Comment:
I don't think this will work. `filename` here is the original file name from Commons, not a path on disk. `data` keeps the actual image file. 

If the mastodon API doesn't allow for uploading images from a file object, `data` should be saved to a tempfile and then `media_post` should be called with the temp file name.